src/main/scala/overflowdb/codegen/CodeGen.scala

```diff
@@ -1,27 +1,20 @@
 package overflowdb.codegen
 
 import better.files._
+
 import java.io.{File => JFile}
+import overflowdb.schema._
 
-object CodeGen extends App {
-  assert(args.size == 2, s"expected two arguments (path to schema json file, basePackage), but got ${args.length}")
-  val schemaFile :: basePackage :: Nil = args.toList
-  val outputDir = new java.io.File("target")
-  new CodeGen(schemaFile, basePackage).run(outputDir)
-}
+import scala.collection.mutable
 
-/** Generates a domain model for OverflowDb traversals based on your domain-specific json schema.
-  *
-  * @param schemaFile: path to the schema (json file)
-  * @param basePackage: specific for your domain, e.g. `com.example.mydomain`
-  */
-class CodeGen(schemaFile: String, basePackage: String) {
+/** Generates a domain model for OverflowDb traversals based on your domain-specific schema. */
+class CodeGen(schema: Schema) {
   import Helpers._
+  val basePackage = schema.basePackage
   val nodesPackage = s"$basePackage.nodes"
   val edgesPackage = s"$basePackage.edges"
-  val schema = new Schema(schemaFile)
 
-  def run(outputDir: JFile): List[JFile] =
+  def run(outputDir: JFile): Seq[JFile] =
```

Comment:
Maybe we could use  `better.files.File` consistently everywhere here?

Reply:
I'd like to keep the public api (just `.run` really) as simple as possible, hence using java File. However I'll make everything else protected and use better.files internally consistently.

Reply:
done